phoenix-core/src/main/java/org/apache/phoenix/query/QueryConstants.java

```diff
@@ -138,6 +138,10 @@
     public final static String SYSTEM_SCHEMA_NAME = "SYSTEM";
     public final static byte[] SYSTEM_SCHEMA_NAME_BYTES = Bytes.toBytes(SYSTEM_SCHEMA_NAME);
     public final static String PHOENIX_METADATA = "table";
+    public final static String SCAN_OFFSET = "OFFSET";
```

Comment:
Please change this to "_RowOffset" (to make is a little less general) and move to BaseScannerRegionObserver (this is where we keep all attributes we set on the Scan so we hopefully notice if we have a duplicate).